install.sh

```diff
@@ -178,8 +183,69 @@ legacy_transformations_end() {
 
     # Remove legacy registration file (we moved it to another location)
     if [ -f "$INSTALL_DIR/autoupdate/registered" ]; then
-        rm -rf "$INSTALL_DIR/autoupdate"
+        rm -rf "$INSTALL_DIR/autoupdate" || LEGACY_TRANSFORM_FAILURES="true"
+    fi
+
+    legacy_transformations_split_global_shared_entries || LEGACY_TRANSFORM_FAILURES="true"
+
+    if [ "$LEGACY_TRANSFORM_FAILURES" = "true" ]; then
+        echo "! There were legacy transform errors: check stderr"
+        return 1
+    fi
+
+    return 0
+}
+
+############################################################
+# Transform all comma-delimited global \`githooks.shared\`
+#   values into multiple git config values.
+#
+# Returns:
+#   1 when failed, 0 otherwise
+############################################################
+legacy_transformations_split_global_shared_entries() {
+
+    CURRENT_LIST=$(git config --global --get-all githooks.shared)
+
+    # If we have more then one of these githooks.shared values
+    # we have passed this install update already.
+    if [ "$(echo "$CURRENT_LIST" | wc -l)" != "1" ]; then
+        return
     fi
+
+    FAILURE="false"
+
+    # If it contains a comma, split it...
+    if echo "$CURRENT_LIST" | grep ","; then
+
+        git config --global --unset githooks.shared
+
+        # Split it and add all back
+        IFS="$IFS_COMMA_NEWLINE"
+        for ITEM in $CURRENT_LIST; do
+            unset IFS
+
+            if [ -n "$ITEM" ]; then
+                git config --global --add githooks.shared "$ITEM" || FAILURE="true"
+            fi
+
+            IFS="$IFS_COMMA_NEWLINE"
+        done
+        unset IFS
+
+    else
+        git config --global --add githooks.shared "$CURRENT_LIST" || FAILURE="true"
```

Comment:
This is not needed, right? If there are no commas, the value can be left unchanged? Now I think it will be duplicated?

Reply:
We make it `--unset` above, to avoid using a `name [value_regex]`  of `git config` which is problematic in case of URLS (needs escaping an stuff like that)

Reply:
But the unset is *inside* the if, not above the if, so that never runs at the same time with this else.

Reply:
Oh crap good catch! Its good we do Reviews... Thx